proposals/apple-mdm/base-setup-clix.md

```diff
@@ -0,0 +1,176 @@
+# MDM Base Setup
+
+Note: these can all be namespaced behind something like `apple-mdm` if needed. But I would like to avoid creating an mdm vertical vs. an osquery vertical to create an integrated product.
+
+## APNS
+### Step 1 - Generate CSR
+```
+$ fleetctl setup apns init
+Generating APNS keys at ./
+  fleet-apns-pki.crt ... done
+  fleet-apns-pki.key ... done
+  fleet-apns-push.key ... done
+  fleet-apns-push.csr ... done
+Send the .csr and .crt files to your friendly Fleet representative
+
+Keep the .key files. You will need them to complete this process. 
+
+Fleet will return a .p7m file to you within (TODO) business days. Use that file in the `fleetctl setup apns gen-req` command.
+```
+
+### Step 2 - Generate .req file from .p7m
+```
+$ fleetctl setup apns gen-req \
+--p7-file=~/path/to/*.p7 \
+--pki-key=~/path/to/fleet-apns-pki.key \
+--pki-crt=~/path/to/fleet-apns-pki.crt
+Working...
+Generated fleet-apns-push.req file at ./
+
+Visit identity.apple.com to upload this .req file. For more details, follow these instructions: NEED LINK HERE
+
+Apple will generate a .crt file. Use that file in the `fleetctl setup apns finalize` command. 
+```
+
+### Step 3 - Upload push..crt and push.key files to Fleet ENV
+
+```sh
+$ fleetctl setup apns finalize
+Update your Fleet server's environment variables as follows:
+1. Update ENV["APPLE_MDM_ENABLE"]=1
+2. Update ENV["APNS_PUSH_CERT"] with the contents of the Apple .crt file
+3. Update ENV["APNS_PUSH_KEY"] with the contents of fleet-apns-push.key
+
+Restart the server to apply the new ENV variables.
+
+```
+
+## SCEP
+### Step 1 - Generate SCEP CA keys
+```
+$ fleetctl setup scep-ca init
+Generating SCEP CA and keys at ./
+  fleet-scep.crt ... done
+  fleet-scep.key ... done
+  
+Next, use these file in the `fleetctl setup scep-ca finalize` command.
+```
+
+### Step 2 - Update fleet server ENV vars with new certs
+```
+$ fleetctl setup scep-ca finalize
```

Comment:
@lucasmrod Could someone set up only one of apns or scep and then go ahead and start enrolling their devices? What happens if so?

Since apns and scep are separate commands, I can imagine people setting up APNS and then forgetting about SCEP.

Reply:
I tried to address this by adding a `fleetctl setup apple-mdm` command that gives the full list of instructions. See below. WDYT?

Reply:
> Could someone set up only one of apns or scep and then go ahead and start enrolling their devices? What happens if so?

`fleet serve` would fail right away if the user does not provide the APNS cert+key + SCEP cert+key.